<a href="https://colab.research.google.com/github/IzabelaSanita/Praticando-Python/blob/main/Praticando_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Sobre o notebook

A "hora da prática" é uma atividade do curso de Python para Data Science da Alura que contém diversos desafios envolvendo os aprendizados de cada aula.

O notebook "hora da pratica" é um notebook construído com espaço para as células dos desafios propostos no curso, com ele você tem espaço para construir e executar suas soluções

## Aula 1 - Introdução ao Python

### Questão 1

Imprima a frase `Escola de Dados da Alura!`

In [ ]:
print('Escola de Dados da Alura!')

Escola de Dados da Alura!


### Questão 2
Imprima na tela seu nome e seu sobrenome seguindo a estrutura abaixo:
```
Nome: [seu nome]
Sobrenome: [seu sobrenome]
```


In [ ]:
print('Nome: Izabela')
print('Sobrenome: Sanitá')

Nome: Izabela
Sobrenome: Sanitá


### Questão 3
Imprima o seu primeiro nome letra a letra. Por exemplo, nome é Mirla, então eu obtenho a seguinte saída:
```
M
I
R
L
A
```


In [ ]:
print('I')
print('Z')
print('A')
print('B')
print('E')
print('L')
print('A')

I
Z
A
B
E
L
A


### Questão 4
Imprima o dia do seu nascimento em formato `dia mês ano`. Os valores de dia e ano não podem estar entre aspas. O formato deve estar como no exemplo, supondo uma data de aniversário dia 28 de fevereiro de 2003:
```
28 fevereiro 2003
```

In [ ]:
print(4, 'setembro', 2007)

4 setembro 2007


### Questão 5
Imprima em um único print o atual ano que você está fazendo esse curso, o valor do ano deve ser um dado numérico. A saída do print deve ser a seguinte:
```
Ano atual: [ano]
```


In [ ]:
print('Ano atual:', 2025)

Ano atual: 2025


## Aula 2 - Manipulando dados no Python

### Coleta e amostragem de dados

#### Questão 1
Crie um programa que solicite à pessoa usuária digitar seu nome, e imprima “Olá, [nome]!”.

In [ ]:
nome = input('Digite seu nome: ')
print(f'Olá,{nome}!')

Digite seu nome: Izabela
Olá,Izabela!


#### Questão 2
Crie um programa que solicite à pessoa usuária digitar seu nome e idade, e imprima “Olá, [nome], você tem [idade] anos.”.

In [ ]:
nome = input('Digite seu nome: ')
idade = int(input('Digite sua idade: '))
print(f'Olá, {nome}, você tem {idade} anos.')

Digite seu nome: Izabela
Digite sua idade: 17
Olá, Izabela, você tem 17 anos.


#### Questão 3
Crie um programa que solicite à pessoa usuária digitar seu nome, idade e altura em metros, e imprima “Olá, [nome], você tem [idade] anos e mede [altura] metros!”.

In [ ]:
nome = input('Digite seu nome: ')
idade = int(input('Digite sua idade: '))
altura = float(input('Digite sua altura em metros: '))
print('Olá, %s, você tem %d anos e mede %.2f metros!' %(nome, idade, altura))

Digite seu nome: Izabela
Digite sua idade: 17
Digite sua altura em metros: 1.50
Olá, Izabela, você tem 17 anos e mede 1.50 metros!


### Calculadora com operadores

#### Questão 1

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a soma dos dois valores.

In [ ]:
valor_1 = int(input('Digite o primeiro valor: '))
valor_2 = int(input('Digite o segundo valor: '))
soma = valor_1 + valor_2
print('O resultado é:', soma)

Digite o primeiro valor: 687
Digite o segundo valor: 89
O resultado é: 776


#### Questão 2
Crie um programa que solicite três valores numéricos à pessoa usuária e imprima a soma dos três valores.

In [ ]:
valor1 = int(input('Digite o primeiro valor: '))
valor2 = int(input('DIgite o segundo valor: '))
valor3 = int(input('Digite o terceiro valor: '))
soma = valor1+valor2+valor3
print('O resultado é:4',soma)

Digite o primeiro valor: 89 
DIgite o segundo valor: 6
Digite o terceiro valor: 4
O resultado é: 99


#### Questão 3
Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a subtração do primeiro pelo o segundo valor.

In [ ]:
v1 = int(input('Primeiro valor: '))
v2 = int(input('Segundo valor: '))
subtracao = v1 - v2
print('O resultado é:', subtracao)

Primeiro valor: 5
Segundo valor: 95
O resultado é: -90


#### Questão 4

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a multiplicação dos dois valores.


In [ ]:
val1 = int(input('Valor um: '))
val2 = int(input('Valor dois: '))
produto = val1 * val2
print('Resultado:', produto)

Valor um: 54
Valor dois: 6
Resultado: 324


#### Questão 5

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e realize a divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.


In [ ]:
num = float(input('Digite um numerador: '))
den = float(input('Digite um denominador que seja diferente de 0: '))
quo = num/den
print('Resultado:', quo)

Digite um numerador: 80
Digite um denominador que seja diferente de 0: 45
Resultado: 1.7777777777777777


#### Questão 6

Crie um programa que solicite dois valores numéricos, um operador e uma potência, e realize a exponenciação entre esses dois valores.


In [ ]:
base = int(input('Digite a base: '))
expoente = int(input('Digite o expoente: '))
potencia = base**expoente
print('Resultado:', potencia)

Número de baixo: 84
Número de cima: 9
Essa merda deu: 208215748530929664


#### Questão 7
Crie um programa que solicite dois valores numéricos, um numerador e um denominador e realize a divisão inteira entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [ ]:
numerador = int(input('Numerador: '))
denominador = int(input('Denominador (diferente de zero): '))
quocienteInteiro = numerador//denominador
print('Resultado:',quocienteInteiro)

Numerador: 5
Denominador (diferente de zero): 2
Resultado: 2


#### Questão 8

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e retorne o resto da divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [ ]:
num = int(input('Numerador: '))
den = int(input('Denominador (diferente de zero): '))
resto = num%den
print('Resultado:',resto)

Numerador: 5
Denominador (diferente de zero): 8
Resultado: 5


#### Questão 9

Crie um código que solicita 3 notas de um estudante e imprima a média das notas.

In [ ]:
nota1 = float(input('nota: '))
nota2 = float(input('nota: '))
nota3 = float(input('nota: '))
print(f'Média: {(nota1+nota2+nota3)/3}')

nota: 2.4
nota: 9.9
nota: 8.2
Média: 6.833333333333333


#### Questão 10

Crie um código que calcule e imprima a média ponderada dos números 5, 12, 20 e 15 com pesos respectivamente iguais a 1, 2, 3 e 4.

In [ ]:
print((5+12*2+20*3+15*4)/10)

14.9


### Editando textos

#### Questão 1

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase na tela.

In [ ]:
frase = 'Izabela Sanitá'
print(frase)

Izabela Sanitá


#### Questão 2
Crie um código que solicite uma frase e depois imprima a frase na tela.

In [ ]:
frase = input('Digite sua frase: ')
print(frase)

Digite sua frase: Izabela
Izabela


#### Questão 3

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras maiúsculas.

In [ ]:
frase = input('Escreva aqui: ')
print(frase.upper())

Escreva aqui: izabela
IZABELA


#### Questão 4

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras minúsculas.

In [ ]:
frase = input('Digite em letras maiúsculas: ')
print(frase.lower())

Digite em letras maiúsculas: IZABELA
izabela


#### Questão 5

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase sem espaços em branco no início e no fim.

In [ ]:
frase = '          iza           '
print(frase.strip())

iza


#### Questão 6

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim.

In [ ]:
frase = input('Digite sua frase: ')
print(frase.strip())

Digite sua frase:                   iza               
iza


#### Questão 7


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim e em letras minúsculas.

In [ ]:
frase = input('D[]igite algo: ')
print(frase.strip().lower())

Digite algo:                        IZA            
iza


#### Questão 8

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`e`” trocadas pela letra “`f`”.

In [ ]:
frase = input('Digite algo que tenha a letra "E": ')
print(frase.replace('e','f'))

Digite algo: Meu nome é Izabela
Mfu nomf é Izabfla


#### Questão 9


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`a`” trocadas pela caractere  “`@`”.

In [ ]:
arroba = input('Digite uma frase que contenha a letra "A": ')
print(arroba.replace('a', chr(64)))

Digite uma frase que contenha a letra "A": Izabela
Iz@bel@


#### Questão 10


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as consoantes “`s`” trocadas pelo caractere  “`$`”.

In [ ]:
frase = input('Digite algo com a letra "S": ')
print(frase.replace('s', chr(36)))

Digite algo com a letra "S": sanitá
$anitá


## Aula 3 - Estruturas condicionais

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça à pessoa usuária para fornecer dois números e exibir o número maior.

In [ ]:
num1 = int(input('Digite o primeiro número: '))
num2 = int(input('Digite o segundo número: '))
if num1 > num2:
  print(f'{num1} é maior')
elif num1 < num2:
  print(f'{num2} é maior')
else:
  print('São iguais')

Digite o primeiro número: 55
Digite o segundo número: 8
55 é maior


#### Questão 2
Escreva um programa que solicite o percentual de crescimento de produção de uma empresa e informe se houve um crescimento (porcentagem positiva) ou decrescimento (porcentagem negativa).

In [ ]:
per = float(input('Digite o percentual de crescimento de produção: '))

if per > 0:
  print(f'Houve um crescimento de {per}%')
elif per < 0:
  print(f'Houve um descrescimento de {per}%')
else:
  print('Não houve alteração')

Digite o percentual de crescimento de produção: -42.8
Houve um descrescimento de -42.8%


#### Questão 3
Escreva um programa que determine se uma letra fornecida pela pessoa usuária é uma vogal ou consoante.

In [ ]:
letra = input('Digite uma letra: ').lower()
vogais = 'aeiou'

if letra in vogais:
  print(f'A letra "{letra}" é uma vogal')
else:
  print(f'A letra "{letra}" é uma consoante')

Digite uma letra: I
A letra "i" é uma vogal


#### Questão 4
Escreva um programa que leia valores médios de preços de um modelo de carro por 3 anos consecutivos e exiba o valor mais alto e mais baixo entre esses três anos.

In [ ]:
valor1 = float(input('Digite o valor do carro no primeiro ano: '))
valor2 = float(input('Digite o valor do carro no segundo ano: '))
valor3 = float(input('Digite o valor do carro no terceiro ano: '))

maior = valor1
if valor2 > maior:
  maior = valor2
if valor3 > maior:
  maior = valor3

menor = valor1
if valor2 < menor:
  menor = valor2
if valor3 < menor:
  menor = valor3

print(f'O maior valor é {maior}')
print(f'O menor valor é {menor}')

Digite o valor do carro no primeiro ano: 5614
Digite o valor do carro no segundo ano: 65149
Digite o valor do carro no terceiro ano: 16684
O maior valor é 65149.0
O menor valor é 5614.0


#### Questão 5
Escreva um programa que pergunte sobre o preço de três produtos e indique qual é o produto mais barato para comprar.

In [ ]:
p1 = float(input('Qual o preço do primeiro produto? '))
p2 = float(input('Qual o preço do segundo produto? '))
p3 = float(input('Qual o preço do terceiro produto? '))

if p1 < p2 and p1 < p3:
  print('O primeiro produto é mais barato')
elif p2 < p1 and p2 < p3:
  print(' O segundo produto é mais barato')
elif p3 < p1 and p3 < p2:
  print('O terceiro produto é mais barato')
elif p1 == p2 == p3:
  print('Os três produtos possuem o mesmo valor')
else:
  if p1 == p2:
    print('O primeiro e segundo produto são os mais baratos')
  elif p1 == p3:
    print('O primeiro e o terceiro produto são os mais baratos')
  elif p2 == p3:
    print('O segundo e o terceiro produto são os mais baratos')

Qual o preço do primeiro produto? 3
Qual o preço do segundo produto? 1
Qual o preço do terceiro produto? 1
O segundo e o terceiro produto são os mais baratos


#### Questão 6
Escreva um programa que leia três números e os exiba em ordem decrescente.

In [ ]:
num1 = int(input('Digite um número: '))
num2 = int(input('Digite um número: '))
num3 = int(input('Digite um número: '))

if num1 >= num2 and num1 >= num3:
  print(num1)
  if num2 >= num3:
    print(num2)
    print(num3)
  else:
    print(num3)
    print(num2)
elif num2 >= num1 and num2 >= num3:
  print(num2)
  if num1 >= num3:
    print(num1)
    print(num3)
  else:
    print(num3)
    print(num1)
else:
  print(num3)
  if num1 >= num2:
    print(num1)
    print(num2)
  else:
    print(num2)
    print(num1)

Digite um número: 89
Digite um número: 456
Digite um número: 4
456
89
4


#### Questão 7
Escreva um programa que pergunte em qual turno a pessoa usuária estuda ("manhã", "tarde" ou "noite") e exiba a mensagem "Bom Dia!", "Boa Tarde!", "Boa Noite!", ou "Valor Inválido!", conforme o caso.

In [ ]:
turno = input('Em qual turno você estuda? ').lower()
manha = ['manhã', 'matutino', 'manha']
tarde = ['tarde', 'vespertino']
noite = ['noite', 'noturno']

if turno in manha:
  print('Bom Dia!')
elif turno in tarde:
  print('Boa Tarde!')
elif turno in noite:
  print('Boa Noite!')
else:
  print('Valor Inválido!')

Em qual turno você estuda? Manha
Bom Dia!


#### Questão 8
Escreva um programa que peça um número inteiro à pessoa usuária e determine se ele é par ou ímpar. Dica: Você pode utilizar o operador módulo `%`.

In [ ]:
num = int(input('Digite um número inteiro: '))

if (num % 2) == 0:
  print(f'O número {num} é par')
else:
  print(f'O número {num} é ímpar')

Digite um número inteiro: 3
O número 3 é ímpar


#### Questão 9
Escreva um programa que peça um número à pessoa usuária e informe se ele é inteiro ou decimal.

In [ ]:
num = float(input('Digite um número inteiro ou decimal: '))

if num % 1 == 0:
  print('O número é inteiro')
else:
  print('O número é decimal')

Digite um número inteiro ou decimal: 5.6
O número é decimal


### Momento dos projetos

#### Questão 10
Um programa deve ser escrito para ler dois números e, em seguida, perguntar à pessoa usuária qual operação ele deseja realizar. O resultado da operação deve incluir informações sobre o número - se é par ou ímpar, positivo ou negativo e inteiro ou decimal.

In [ ]:
num1 = float(input('Digite o primeiro número: '))
num2 = float(input('Digite o segundo número: '))
op = input('Qual operação deseja realizar? (tente soma, subtração, divisão, multiplicação ou exponenciação): ').lower()
resultado = num1 + num2

if op == 'soma':
  print(f'O resultado é {resultado}')
elif op == 'subtração':
  resultado = num1 - num2
  print(f'O resultado é {resultado}')
elif op == 'multiplicação':
  resultado = num1 * num2
  print(f'O resultado é {resultado}')
elif op == 'divisão':
  resultado = num1 / num2
  print(f'O resultado é {resultado}')
elif op == 'exponenciação':
  resultado = num1 ** num2
  print(f'O resultado é {resultado}')
else:
  resultado = 0
  print('Operação inválida! O resultado será igual a 0')

if resultado % 2 == 0:
  print('É par')
else:
  print('É ímpar')

if resultado >= 0:
  print('É positivo')
else:
  print('É negativo')

if resultado % 1 == 0:
  print('É inteiro')
else:
  print('É decimal')

Digite o primeiro número: 5
Digite o segundo número: 6
Qual operação deseja realizar? (tente soma, subtração, divisão, multiplicação ou exponenciação): fa
Operação inválida! O resultado será igual a 0
É par
É positivo
É inteiro


#### Questão 11
Escreva um programa que peça à pessoa usuária três números que representam os lados de um triângulo. O programa deve informar se os valores podem ser utilizados para formar um triângulo e, caso afirmativo, se ele é equilátero, isósceles ou escaleno. Tenha em mente algumas dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;


In [ ]:
l1 = int(input('Digite o valor do primeiro lado: '))
l2 = int(input('Digite o valor do segundo lado: '))
l3 = int(input('Digite o valor do terceiro lado: '))

if (l1 + l2) > l3 or (l1 + l3) > l2 or (l2 + l3) > l1:
  print('Valor válido!')

  if l1 == l2 == l3:
    print('É um triângulo equilátero')
  elif l1 == l2 or l1 == l3 or l2 ==l3:
    print('É um triângulo isósceles')
  elif l1 != l2 != l3:
    print('É um triângulo escaleno')
else:
  print('Valor inválido!')

Digite o valor do primeiro lado: 5486
Digite o valor do segundo lado: 4
Digite o valor do terceiro lado: 68
Valor válido!
É um triângulo escaleno


#### Questão 12
Um estabelecimento está vendendo combustíveis com descontos variados. Para o etanol, se a quantidade comprada for até 15 litros, o desconto será de 2% por litro. Caso contrário, será de 4% por litro. Para o diesel, se a quantidade comprada for até 15 litros, o desconto será de 3% por litro. Caso contrário, será de 5% por litro. O preço do litro de diesel é R\$ 2,00 e o preço do litro de etanol é R\$ 1,70. Escreva um programa que leia a quantidade de litros vendidos e o tipo de combustível (E para etanol e D para diesel) e calcule o valor a ser pago pelo cliente. Tenha em mente algumas dicas:

- O do valor do desconto será a multiplicação entre preço do litro, quantidade de litros e o valor do desconto.
- O valor a ser pago por um cliente será o resultado da multiplicação do preço do litro pela quantidade de litros menos o valor de desconto resultante do cálculo.


In [ ]:
tipo = input('Qual o tipo de combustível que deseja aplicar ("E" para etanol e "D" para diesel): ').upper()
litros = int(input('Quantos litros deseja aplicar: '))
valor_pago = 0
diesel = 2
etanol = 1.7

if tipo == 'E':
  if litros > 0 and litros < 15:
    valor_pago = (etanol * litros) - (etanol * litros * 0.02)
    print(f'O valor a ser pago é de R${valor_pago}')
  elif litros >= 15:
    valor_pago = (etanol * litros) - (etanol * litros * 0.04)
    print(f'O valor a ser pago é de R${valor_pago}')
  else:
    print('Veio aqui pra quê?')
elif tipo == 'D':
  if litros > 0 and litros < 15:
    valor_pago = (diesel * litros) - (diesel * litros * 0.03)
    print(f'O valor a ser pago é de R${valor_pago}')
  elif litros >= 15:
    valor_pago = (diesel * litros) - (diesel * litros * 0.05)
    print(f'O valor a ser pago é de R${valor_pago}')
  else:
    print('Veio aqui pra quê?')
else:
  print('Não temos essa opção.')


Qual o tipo de combustível que deseja aplicar ("E" para etanol e "D" para diesel): d
Quantos litros deseja aplicar: 20
O valor a ser pago é de R$38.0


#### Questão 13
Em uma empresa de venda de imóveis você precisa criar um código que analise os dados de vendas anuais para ajudar a diretoria na tomada de decisão. O código precisa coletar os dados de quantidade de venda durante os anos de 2022 e 2023 e fazer um cálculo de variação percentual. A partir do valor da variação, deve ser enviada às seguintes sugestões:
- Para variação acima de 20%: bonificação para o time de vendas.
- Para variação entre 2% e 20%: pequena bonificação para time de vendas.
- Para variação entre 2% e -10%: planejamento de políticas de incentivo às vendas.
- Para bonificações abaixo de -10%: corte de gastos.


In [ ]:
vendas22 = float(input('Quantidade de vendas no ano de 2022: '))
vendas23 = float(input('Quantidade de vendas no ano de 2023: '))
variacao = vendas23 - vendas22
variacao = (variacao * 100) / vendas22

if variacao > 20:
  print('O time de vendas terá uma bonificação!')
elif variacao > 2 and variacao >= 20:
  print('O time de vendas terá uma pequena bonificação!')
elif variacao > -10 and variacao <= 2:
  print('Teremos um planejamento de políticas de incentivo às vendas.')
else:
  print('Teremos corte de gastos!')

Quantidade de vendas no ano de 2022: 120
Quantidade de vendas no ano de 2023: 2
Teremos corte de gastos!


## Aula 4 - Estruturas de repetição

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça dois números inteiros e imprima todos os números inteiros entre eles.

In [ ]:
num1 = int(input('Digite o primeiro número: '))
num2 = int(input('Digite o segundo número: '))

if num1 < num2:
  for i in range(num1 + 1, num2):
    print(i)
elif num1 > num2:
  for i in range(num2 + 1, num1):
    print(i)
else:
  print('Os números são iguais!')

Digite o primeiro número: 10
Digite o segundo número: 2
3
4
5
6
7
8
9


#### Questão 2
Escreva um programa para calcular quantos dias levará para a colônia de uma bactéria A ultrapassar ou igualar a colônia de uma bactéria B, com base nas taxas de crescimento de 3% e 1,5% respectivamente. Considere que a colônia A inicia com 4 elementos e a colônia B com 10 elementos.

In [ ]:
bacteria_a = 4
bacteria_b = 10
dias = 0

while bacteria_b > bacteria_a:
  bacteria_a *= 1 + 0.03
  bacteria_b *= 1 + 0.015
  dias +=1
print(f'A bactéria A levou {dias} dias para ultrapassar ou igualar a bactéria B')

A bactéria A levou 63 dias para ultrapassar ou igualar a bactéria B


#### Questão 3
Para tratar uma quantidade de 15 dados de avaliações de pessoas usuárias de um serviço da empresa, precisamos verificar se as notas são válidas. Então, escreva um programa que vai receber a nota de 0 a 5 de todos os dados e verificar se é um valor válido. Caso seja inserido uma nota acima de 5 ou abaixo de 0, repita até que a pessoa usuária insira um valor válido.

In [ ]:
for i in range(15):
  avaliacao = float(input(f'Usuário {i}, insira sua nota de 0 a 5: '))
  while avaliacao < 0 or avaliacao > 5:
    avaliacao = float(input('Nota inválida!Tente novamente com uma nota entre 0 e 5: '))
print('Tudo certo! Notas válidas.')

Usuário 0, insira sua nota de 0 a 5: 1
Usuário 1, insira sua nota de 0 a 5: 1
Usuário 2, insira sua nota de 0 a 5: 1
Usuário 3, insira sua nota de 0 a 5: 10
Nota inválida!Tente novamente com uma nota entre 0 e 5: 2
Usuário 4, insira sua nota de 0 a 5: 2
Usuário 5, insira sua nota de 0 a 5: 2
Usuário 6, insira sua nota de 0 a 5: 2
Usuário 7, insira sua nota de 0 a 5: 2
Usuário 8, insira sua nota de 0 a 5: 2
Usuário 9, insira sua nota de 0 a 5: 2
Usuário 10, insira sua nota de 0 a 5: 2
Usuário 11, insira sua nota de 0 a 5: 2
Usuário 12, insira sua nota de 0 a 5: 2
Usuário 13, insira sua nota de 0 a 5: 2
Usuário 14, insira sua nota de 0 a 5: 2
Tudo certo! Notas válidas.


#### Questão 4
Desenvolva um programa que leia um conjunto indeterminado de temperaturas em Celsius e faça uma análise. Portanto, escreva um programa que leia temperaturas e informe a média delas. A leitura deve ser encerrada ao ser enviado o valor -273°C.

In [ ]:
temperatura = float(input('Digite a temperatura em Celsius: '))

contadora = 0
soma = 0

while temperatura != -273:
  soma += temperatura
  contadora += 1
  temperatura = float(input('Digite a temperatura em Celsius: '))

media = soma/contadora
print(f'A média das temperaturas é {media}')

Digite a temperatura em Celsius: 2
Digite a temperatura em Celsius: 5
Digite a temperatura em Celsius: 6
Digite a temperatura em Celsius: -273
A média das temperaturas é 4.333333333333333


#### Questão 5
Escreva um programa que calcule o fatorial de um número inteiro fornecido pela pessoa usuária. O fatorial de um número inteiro é a multiplicação desse número por todos os seus antecessores até o número 1. Por exemplo, o fatorial de 5 é 5 x 4 x 3 x 2 x 1 = 120.

In [ ]:
num = int(input('Digite um número inteiro: '))

i = num - 1

while i > 1:
    num = num * i
    i -= 1
print(f'O fatorial é {num}')

Digite um número inteiro: 5
O fatorial é 120


### Momento dos projetos

#### Questão 6
Escreva um programa que gere a tabuada de um número inteiro de 1 a 10, de acordo com a escolha da pessoa usuária. Como exemplo, para o número 2, a tabuada deve ser mostrada no seguinte formato:
```
Tabuada do 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20
```

In [ ]:
num = int(input('Digite um número inteiro: '))

print(f'Tabuada do {num}:')
for tabuada in range (1,11):
  print(f'{num} x {tabuada} = {(num * tabuada)}')
  tabuada += 1

Digite um número inteiro: 5
Tabuada do 5:
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


#### Questão 7
Os números primos possuem várias aplicações dentro da Ciência de Dados, por exemplo, na criptografia e segurança. Um número primo é aquele que é divisível apenas por um e por ele mesmo. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.

In [ ]:
num = int(input('Digite um número inteiro: '))

primo = True

if num <= 1:
  primo = False
else:
  for i in range (2,num):
    if num % i == 0:
      primo = False
      break
if primo:
  print(f'O número {num} é primo')
else:
  print(f'O número {num} não é primo')

Digite um número inteiro: 17
O número 17 é primo


#### Questão 8
Vamos entender a distribuição de idades de pensionistas de uma empresa de previdência. Escreva um programa que leia as idades de uma quantidade não informada de clientes e mostre a distribuição em intervalos de [0-25], [26-50], [51-75] e [76-100]. Encerre a entrada de dados com um número negativo.

In [ ]:
idade = int(input('Digite sua idade: '))
primeiroIntervalo = 0
segundoIntervalo = 0
terceiroIntervalo = 0
quartoIntervalo = 0

while idade <= 100:
  if idade >= 0 and idade <=25:
    primeiroIntervalo += 1
  elif idade >= 26 and idade <= 50:
    segundoIntervalo += 1
  elif idade >= 51 and idade <= 75:
    terceiroIntervalo += 1
  elif idade >= 76 and idade <= 100:
    quartoIntervalo += 1
  else:
    break
  idade = int(input('Digite sua idade:'))
if idade > 100:
  print('Idade inválida')

print(f'Distribuição das idades em intervalos:\n[0-25]: {primeiroIntervalo}\n[26-50]: {segundoIntervalo}\n[51-75]: {terceiroIntervalo}\n[76-100]: {quartoIntervalo}')



Digite sua idade: 15
Digite sua idade:65
Digite sua idade:95
Digite sua idade:42
Digite sua idade:6
Digite sua idade:8
Digite sua idade:41
Digite sua idade:-1
Distribuição das idades em intervalos:
[0-25]: 3
[26-50]: 2
[51-75]: 1
[76-100]: 1


#### Questão 9
Em uma eleição para gerência em uma empresa com 20 funcionários, existem quatro candidatos. Escreva um programa que calcule o vencedor da eleição. A votação ocorreu da seguinte maneira:
- Cada funcionário votou em um dos quatro candidatos (representados pelos números 1, 2, 3 e 4).
- Também foram contabilizados os votos nulos (representado pelo número 5) e os votos em branco (representado pelo número 6).

Ao final da votação, o programa deve exibir o total de votos para cada candidato, o número de votos nulos e o número de votos em branco. Além disso, deve calcular e exibir a porcentagem de votos nulos em relação ao total de votos e a porcentagem de votos em branco em relação ao total de votos.

In [ ]:
candidato1 = 0
candidato2 = 0
candidato3 = 0
candidato4 = 0
nulo = 0
branco = 0

for i in range(1, 21):
  voto = int(input(f'Funcionário {i}, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): '))
  if voto == 1:
    candidato1 += 1
  elif voto == 2:
    candidato2 += 1
  elif voto == 3:
    candidato3 +=1
  elif voto == 4:
    candidato4 += 1
  elif voto == 5:
    nulo += 1
  elif voto == 6:
    branco += 1
  else:
    print('Voto inválido!')

print(f'Total:\nCandidato 1: {candidato1}\nCandidato 2: {candidato2}\nCandidato 3 : {candidato3}\nCandidato 4: {candidato4}\nVotos Nulos: {nulo}\nVotos em Branco: {branco}')
print(f'Percentual de votos nulos: {(nulo / 20 * 100)}')
print(f'Percentual de votos em branco: {(branco / 20 * 100)}')


Funcionário 1, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 5
Funcionário 2, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 5
Funcionário 3, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 6
Funcionário 4, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 6
Funcionário 5, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 1
Funcionário 6, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 2
Funcionário 7, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 3
Funcionário 8, digite sua opção (1, 2, 3 e 4 para os respectivos candidatos, 5 para votar nulo e 6 para votar em branco): 4
Funcioná

## Aula 5 - Estruturas de dados

### Aquecendo na programação

#### Questão 1
Faça um programa que tenha a seguinte lista contendo os valores de gastos de uma empresa de papel `[2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]`. Com esses valores, faça um programa que calcule a média de gastos. Dica, use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).

In [ ]:
gastos = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]
somaGastos = sum(gastos)
quantidadeGastos = len(gastos)
mediaGastos = somaGastos / quantidadeGastos

print(f'A média de gastos da empresa é de R${mediaGastos}')

A média de gastos da empresa é de R$3144.222


#### Questão 2
 Com os mesmos dados da questão anterior, defina quantas compras foram acima de 3000 reais e calcule a porcentagem quanto ao total de compras.

In [ ]:
gastos = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]
compras3000 = 0

for valor in gastos:
  if valor > 3000.00:
    compras3000 += 1

porcentagem3000 = (compras3000 * 100) / len(gastos)

print(f'Quantidade de compras acima de 3000 reais: {compras3000}')
print(f'Porcentagem em relação ao total: {porcentagem3000}%')

Quantidade de compras acima de 3000 reais: 7
Porcentagem em relação ao total: 70.0%


#### Questão 3
Faça um código que colete em uma lista 5 números inteiros quaisquer e imprima a lista. Exemplo: `[1,4,7,2,4]`.

In [ ]:
lista = []

for i in range(0, 5):
  numero = int(input('Digite um número inteiro qualquer: '))
  lista.append(numero)
print(f'Sua lista final: {lista}')

Digite um número inteiro qualquer: 1
Digite um número inteiro qualquer: 4
Digite um número inteiro qualquer: 7
Digite um número inteiro qualquer: 2
Digite um número inteiro qualquer: 4
Sua lista final: [1, 4, 7, 2, 4]


#### Questão 4
Colete novamente 5 inteiros e imprima a lista em ordem inversa à enviada.

In [ ]:
lista = []

for i in range(0, 5):
  numero = int(input('Digite um número inteiro qualquer: '))
  lista.insert(0, numero)
print(f'Sua lista final inversa: {lista}')

Digite um número inteiro qualquer: 4
Digite um número inteiro qualquer: 0
Digite um número inteiro qualquer: 0
Digite um número inteiro qualquer: 2
Digite um número inteiro qualquer: 8
Sua lista final inversa: [8, 2, 0, 0, 4]


#### Questão 5
Faça um programa que, ao inserir um número qualquer, criará uma lista contendo todos os números primos entre 1 e o número digitado.

In [ ]:
numero = int(input('Digite um número: '))
lista = []

for num in range(2, numero):
  primo = True

  for teste in range(2, num):
    if num % teste == 0:
      primo = False
      break
  if primo:
    lista.append(num)
print(f'Números primos: {lista}')

Digite um número: 10
Números primos: [2, 3, 5, 7]


#### Questão 6
Escreva um programa que peça uma data informando o dia, mês e ano e determine se ela é uma data válida para uma análise.

In [ ]:
dia = int(input('Digite o dia: '))
mes = int(input('Digite o mês: '))
ano = int(input('Digite o ano: '))

if mes == 2:
  if ano % 4 == 0 and (ano % 400 == 0 or ano % 100 != 0):
    dias_fevereiro = 29
  else:
    dias_fevereiro = 28
  if dia >= 1 and dia <= dias_fevereiro:
    print('Data válida!')
  else:
    print('Data inválida!')
elif mes in [1, 3, 5, 7, 8, 10, 12]:
  if dia >= 1 and dia <= 31:
    print('Data válida!')
  else:
    print('Data inválida!')
elif mes in [2, 4, 6, 9, 11]:
  if dia >= 1 and dia <= 30:
    print('Data válida!')
  else:
    print('Data inválida!')
else:
  print('Data inválida!')

Digite o dia: 29
Digite o mês: 2
Digite o ano: 2016
Data válida!


### Momento dos projetos

#### Questão 7
Para um estudo envolvendo o nível de multiplicação de bactérias em uma colônia, foi coletado o número de bactérias multiplicadas por dia, com base em um processo de duplicação diária e pode ser observado a seguir: `[1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]`. Tendo esses valores, faça um código que gere uma lista contendo o percentual de crescimento de bactérias por dia, comparando o número de bactérias em cada dia com o número de bactérias do dia anterior. Dica: para calcular o percentual de crescimento usamos a seguinte equação: `100 * (amostra_atual - amostra_passada) / (amostra_passada)`.

In [ ]:
amostraBacteria = [1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]
listaPercentual = []

for i in range(1, len(amostraBacteria)):
  percentual = 100 * (amostraBacteria[i] - amostraBacteria[i-1]) / (amostraBacteria[i-1])
  listaPercentual.append(percentual)
print(f'Percentuais de Crescimento:\n{listaPercentual}')

Percentuais de Crescimento:
[75.00000000000001, 57.142857142857125, 51.51515151515153, 56.0, 44.87179487179489, 46.90265486725664, 51.20481927710843, 50.59760956175297, 50.52910052910054]


#### Questão 8
Para uma seleção de produtos alimentícios, precisamos separar o conjunto de IDs dados por números inteiros, sabendo que os produtos com ID par são doces e os com ID ímpar são amargos. Monte um código que colete 10 IDs. Depois, calcule e mostre a quantidade de produtos doces e amargos.

In [ ]:
doces = 0
amargos = 0
listaID = []

for i in range(0, 10):
  listaID.append(int(input(f'Digite o {i+1}° ID: ')))
for id in listaID:
  if id % 2 == 0:
    doces += 1
  else:
    amargos += 1

print(f'Produtos doces: {doces}\nProdutos amargos: {amargos}')

Digite o 1° ID: 56
Digite o 2° ID: 2
Digite o 3° ID: 86
Digite o 4° ID: 7
Digite o 5° ID: 9
Digite o 6° ID: 5
Digite o 7° ID: 25
Digite o 8° ID: 6
Digite o 9° ID: 1
Digite o 10° ID: 5
Produtos doces: 4
Produtos amargos: 6


#### Questão 9
Desenvolva um programa que informa a nota de um aluno de acordo com suas respostas. Ele deve pedir a resposta de um aluno para cada questão e é preciso verificar se a resposta foi igual ao gabarito. Cada questão vale um ponto e existem alternativas:  A, B, C ou D.
```
Gabarito da Prova:
01 - D
02 - A
03 - C
04 - B
05 - A
06 - D
07 - C
08 - C
09 - A
10 - B
```

In [ ]:
gabarito = ['D', 'A', 'C', 'B', 'A', 'D', 'C', 'C', 'A', 'B']
respostas = []
nota = 0

for i in range(0, 10):
  respostas.append(input(f'Digite a resposta da questão {i+1}: ').upper())
for i in range(0, 10):
  if respostas[i] == gabarito[i]:
    nota +=1
print(f'Sua nota é: {nota}')

Digite a resposta da questão 1: a
Digite a resposta da questão 2: a
Digite a resposta da questão 3: a
Digite a resposta da questão 4: a
Digite a resposta da questão 5: a
Digite a resposta da questão 6: a
Digite a resposta da questão 7: a
Digite a resposta da questão 8: a
Digite a resposta da questão 9: a
Digite a resposta da questão 10: a
Sua nota é: 3


#### Questão 10
Um instituto de meteorologia deseja fazer um estudo de temperatura média de cada mês do ano. Para isso, você precisa fazer um código que colete e armazene essas temperaturas médias em uma lista. Depois, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês elas ocorreram, mostrando os meses por extenso: Janeiro, Fevereiro, etc.

In [ ]:
temperaturas = []
meses = ['Janeiro', 'Fevereiro', 'Março', 'Abril', 'Maio', 'Junho', 'Julho', 'Agosto', 'Setembro', 'Outubro', 'Novembro', 'Dezembro']

for i in meses:
  temperaturas.append(float(input(f'Digite a temperatura média do mês de {i}: ')))

mediaAnual = sum(temperaturas) / len(temperaturas)

print(f'Média de temperatura anual: {mediaAnual}°\nMeses em que a temperatura ultrapassou a média anual: ')

for i in range(0, 12):
  if temperaturas[i] > mediaAnual:
    print(meses[i])

Digite a temperatura média do mês de Janeiro: 5
Digite a temperatura média do mês de Fevereiro: 6
Digite a temperatura média do mês de Março: 8
Digite a temperatura média do mês de Abril: 10
Digite a temperatura média do mês de Maio: 25
Digite a temperatura média do mês de Junho: 23
Digite a temperatura média do mês de Julho: 28
Digite a temperatura média do mês de Agosto: 30
Digite a temperatura média do mês de Setembro: 32
Digite a temperatura média do mês de Outubro: 14
Digite a temperatura média do mês de Novembro: 11
Digite a temperatura média do mês de Dezembro: 2
Média de temperatura anual: 16.166666666666668°
Meses em que a temperatura ultrapassou a média anual: 
Maio
Junho
Julho
Agosto
Setembro


#### Questão 11
Uma empresa de e-commerce está interessada em analisar as vendas dos seus produtos. Os dados das vendas foram armazenados em um dicionário:
```
{'Produto A': 300, 'Produto B': 80, 'Produto C': 60,
 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}
```
Escreva um código que calcule o total de vendas e o produto mais vendido.

In [ ]:
dadosVendas = {'Produto A': 300, 'Produto B': 80, 'Produto C': 60, 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}

totalVendas = 0
produtoMaisVendido = ''
unidadesProdutoMaisVendido = 0

for produto in dadosVendas.keys():
  totalVendas += dadosVendas[produto]
  if dadosVendas[produto] > unidadesProdutoMaisVendido:
    unidadesProdutoMaisVendido = dadosVendas[produto]
    produtoMaisVendido = produto

print(f'Total de vendas: {totalVendas}\nProduto mais vendido: {produtoMaisVendido}')

Total de vendas: 920
Produto mais vendido: Produto A


#### Questão 12
Uma pesquisa de mercado foi feita para decidir qual design de uma marca infantil mais agrada crianças. A pesquisa foi feita e o votos computados podem ser observados abaixo:
```
'''
Tabela de votos da marca
Design 1 - 1334 votos
Design 2 - 982 votos
Design 3 - 1751 votos
Design 4 - 210 votos
Design 5 - 1811 votos
'''
```
Adapte os dados fornecidos a você para uma estrutura de dicionário e a partir dele, informe o design vencedor e a porcentagem de votos recebidos.


In [ ]:
votos = {'Design 1': 1334, 'Design 2': 982, 'Design 3': 1751, 'Design 4': 210, 'Design 5': 1811}

designMaisVotado = ''
totalVotos = 0
votosMaisVotado = 0

for design in votos.keys():
  totalVotos += votos[design]

  if votos[design] > votosMaisVotado:
    votosMaisVotado = votos[design]
    designMaisVotado = design

porcentagemVencedor = 100 * (votosMaisVotado) / (totalVotos)

print(f'O {designMaisVotado} venceu com {porcentagemVencedor}% dos votos!')

O Design 5 venceu com 29.747043363994745% dos votos!


#### Questão 13
Os funcionários de um setor da empresa que você trabalha vão receber um abono correspondente a 10% do seu salário devido ao ótimo desempenho do time. O setor financeiro solicitou sua ajuda para a verificação das consequências financeiras que esse abono irá gerar nos recursos. Assim, foi encaminhada para você uma lista com os salários que receberão o abono: `[1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]`. O abono de cada funcionário não pode ser inferior a 200. Em código, transforme cada um dos salários em chaves de um dicionário e o abono de cada salário no elemento. Depois, informe o total de gastos com o abono, quantos funcionários receberam o abono mínimo e qual o maior valor de abono fornecido.


In [ ]:
salarios = [1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]
dicAbonos = {}
totalAbonos = 0
quantidadeAbonoMinimo = 0
maiorAbono = 0

for salario in salarios:
  abono = salario * 0.1
  if abono < 200:
    abono = 200
  dicAbonos[salario] = abono

for abono in dicAbonos.values():
  totalAbonos += abono
  if abono == 200:
    quantidadeAbonoMinimo += 1
  if abono > maiorAbono:
    maiorAbono = abono

print(f'Abonos: {dicAbonos}\nTotal de Gastos com Abono: {totalAbonos}\nQuantidade de Abonos Mínimos: {quantidadeAbonoMinimo}\nMaior Abono: {maiorAbono}')


Abonos: {1172: 200, 1644: 200, 2617: 261.7, 5130: 513.0, 5532: 553.2, 6341: 634.1, 6650: 665.0, 7238: 723.8000000000001, 7685: 768.5, 7782: 778.2, 7903: 790.3000000000001}
Total de Gastos com Abono: 6087.8
Quantidade de Abonos Mínimos: 2
Maior Abono: 790.3000000000001


#### Questão 14
Uma equipe de cientistas de dados está estudando a diversidade biológica em uma floresta. A equipe fez a coleta de informações sobre o número de espécies de plantas e animais em cada área da floresta e armazenaram essas informações em um dicionário. Nele, a chave descreve a área dos dados e os valores nas listas correspondem às espécies de plantas e animais nas áreas, respectivamente.
```
{'Área Norte': [2819, 7236],
 'Área Leste': [1440, 9492],
 'Área Sul': [5969, 7496],
 'Área Oeste': [14446, 49688],
 'Área Centro': [22558, 45148]}
```
Escreva um código para calcular a média de espécies por área e identificar a área com a maior diversidade biológica. Dica: use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).


In [ ]:
diversidadeBiologica = {'Área Norte': [2819, 7236],
 'Área Leste': [1440, 9492],
 'Área Sul': [5969, 7496],
 'Área Oeste': [14446, 49688],
 'Área Centro': [22558, 45148]}

maiorDiversidade = ''
maiorSoma = 0
somaEspecies = 0

for area, especies in diversidadeBiologica.items():
  media = sum(especies) / len(especies)
  somaEspecies = sum(especies)
  print(f'A média de espécies da {area} é igual a {media}.')
  if maiorSoma < somaEspecies:
    maiorSoma = somaEspecies
    maiorDiversidade = area
print(f'A {maiorDiversidade} tem a maior diversidade biológica.')


A média de espécies da Área Norte é igual a 5027.5.
A média de espécies da Área Leste é igual a 5466.0.
A média de espécies da Área Sul é igual a 6732.5.
A média de espécies da Área Oeste é igual a 32067.0.
A média de espécies da Área Centro é igual a 33853.0.
A Área Centro tem a maior diversidade biológica.


#### Questão 15
O setor de RH da sua empresa te pediu uma ajuda para analisar as idades dos funcionários de 4 setores da empresa. Para isso, ele te forneceu os seguintes dados:
```
{'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
```
Sabendo que cada setor tem 10 funcionários, construa um código que calcule a média de idade de cada setor, a idade média geral entre todos os setores e quantas pessoas estão acima da idade média geral.


In [ ]:
idadesSetor = {'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}

mediaSetor = 0
mediaGeral = 0
acimaMedia = 0
listaMedias = []

for setor, idades in idadesSetor.items():
  mediaSetor = sum(idades) / len(idades)
  print(f'A média de idade dos funcionários do {setor} é de {mediaSetor} anos.')
  listaMedias.append(mediaSetor)

mediaGeral = sum(listaMedias) / len(listaMedias)

for idades in idadesSetor.values():
  for id in idades:
    if id > mediaGeral:
      acimaMedia += 1

print(f'Idade média geral: {mediaGeral}\nQuantidade de funcionários com a idade acima da média: {acimaMedia}')



A média de idade dos funcionários do Setor A é de 39.9 anos.
A média de idade dos funcionários do Setor B é de 42.3 anos.
A média de idade dos funcionários do Setor C é de 35.8 anos.
A média de idade dos funcionários do Setor D é de 36.5 anos.
Idade média geral: 38.625
Quantidade de funcionários com a idade acima da média: 18
